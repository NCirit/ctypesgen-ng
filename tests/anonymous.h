#if 0
/*
 * TODO: there's no easy way to implement this with libclang 3.6; trunk adds a
 * feature that will make it possible.
 */
struct tagged1 {
	enum type1 {
		STRING1,
		INT1,
	} t;
	union var1 {
		char *s;
		int i;
	};
};
#endif

struct body {
	struct {
		int x, y;
	} point;
	struct {
		int mass;
		int volume;
	} size;
	union {
		int liquid;
		int gas;
	} state;
};

struct nested {
	struct {
		struct {
			int a;
		} A;
		int b;
	} B;
	int c;
};
