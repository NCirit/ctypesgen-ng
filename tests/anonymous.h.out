import ctypes

libtest = ctypes.CDLL('libtest.so')


class __anon_struct1(ctypes.Structure):
    pass
__anon_struct1._fields_ = [
    ('x', ctypes.c_int),
    ('y', ctypes.c_int),
]


class __anon_struct2(ctypes.Structure):
    pass
__anon_struct2._fields_ = [
    ('mass', ctypes.c_int),
    ('volume', ctypes.c_int),
]


class __anon_union1(ctypes.Union):
    pass
__anon_union1._fields_ = [
    ('liquid', ctypes.c_int),
    ('gas', ctypes.c_int),
]


class struct_body(ctypes.Structure):
    pass
struct_body._fields_ = [
    ('point', __anon_struct1),
    ('size', __anon_struct2),
    ('state', __anon_union1),
]


class __anon_struct3(ctypes.Structure):
    pass
__anon_struct3._fields_ = [
    ('a', ctypes.c_int),
]


class __anon_struct4(ctypes.Structure):
    pass
__anon_struct4._fields_ = [
    ('A', __anon_struct3),
    ('b', ctypes.c_int),
]


class struct_nested(ctypes.Structure):
    pass
struct_nested._fields_ = [
    ('B', __anon_struct4),
    ('c', ctypes.c_int),
]
